src/utils/language.ts

```diff
@@ -1,19 +1,40 @@
-import { detect } from 'tinyld'
+import { detectAll } from 'tinyld'
+import { Video } from '../model'
 
 function cleanString(input: string): string {
-  // Remove symbols, numbers, and emojis
-  // The regex [\p{P}\p{S}\p{N}\p{M}] matches all kinds of punctuation, symbols, numbers, and mark characters (including emojis)
-  // \p{P} matches any kind of punctuation character
-  // \p{S} matches any kind of math symbol, currency sign, or modifier symbol
-  // \p{N} matches any kind of numeric character in any script
-  // \p{M} matches characters that are combined with other characters, often used for emojis and diacritics
-  // The 'u' flag enables Unicode support, allowing the regex to match Unicode characters and properties
-  const cleanedString = input.replace(/[\p{P}\p{S}\p{N}\p{M}]/gu, '')
-  return cleanedString.toLowerCase()
+  // First, remove URLs. This pattern targets a broad range of URLs.
+  let cleanedString = input.replace(/(https?:\/\/[^\s]+)|(www\.[^\s]+)/gu, '')
+
+  // Remove hashtags. This regex looks for the '#' symbol followed by one or more word characters.
+  cleanedString = cleanedString.replace(/#\w+/gu, '')
+
+  return cleanedString
 }
 
-// Example usage
-export const predictLanguage = (text: string): string | undefined => {
+export function predictLanguage(text: string): { lang: string; accuracy: number } | undefined {
   const cleanedText = cleanString(text)
-  return detect(cleanedText) || undefined
+
+  // console.log(`Cleaned text: ${cleanedText}`)
+  // Get the most accurate language prediction
+  return detectAll(cleanedText).length ? detectAll(cleanedText)[0] : undefined
+}
+
+export function predictVideoLanguage({
+  title,
+  description,
+}: Pick<Video, 'title' | 'description'>): string | undefined {
+  let detectedLang: string | undefined
+
+  const titleLang = predictLanguage(title ?? '')
+  if (titleLang && titleLang?.accuracy < 0.5) {
```

Comment:
Did you make some benchmarking at what threshold prediction is mostly wrong?

Reply:
No, It's just a numerical guess, i.e. if the prediction confidence is less than 50%, then do the above logic.